added hint for usaco-788 (#3891)
cpinitiative/usaco-guide: solutions/silver/usaco-788.mdx

author: Óscar Garries, Qi Wang
---

<Spoiler title="Hint 1">
When there are $N$ videos and $N - 1$ relationships between them, what data structure you think can best store and process those information?
</Spoiler>

<Spoiler title="Hint 2">

> Say we only look at one query, imagine we search through a tree of videos manually, notice how the relevance value influences our decision making process in selecting the relevant videos.

Reviewer: ```suggestion
Let's imagine manually searching through a tree of videos for a singular query. What observations can we make about how the relevance value influences our decision-making process in selecting the relevant videos?
```

(optional) 

just thought the original read kinda weirdly...

> [Official Analysis (Java)](http://www.usaco.org/current/data/sol_mootube_silver_jan18.html)

Reviewer 2: put this in the solution spoiler?

Author: oh ok
